sdk/storage/inc/blobs/blob_client.hpp

```diff
@@ -367,12 +367,18 @@ namespace Azure { namespace Storage { namespace Blobs {
   protected:
     UriBuilder m_blobUrl;
     std::shared_ptr<Azure::Core::Http::HttpPipeline> m_pipeline;
+    Azure::Core::Nullable<EncryptionKey> m_customerProvidedKey;
+    Azure::Core::Nullable<std::string> m_encryptionScope;
 
   private:
     explicit BlobClient(
         UriBuilder blobUri,
-        std::shared_ptr<Azure::Core::Http::HttpPipeline> pipeline)
-        : m_blobUrl(std::move(blobUri)), m_pipeline(std::move(pipeline))
+        std::shared_ptr<Azure::Core::Http::HttpPipeline> pipeline,
+        Azure::Core::Nullable<EncryptionKey> customerProvidedKey,
```

Comment:
Should these just be in the options struct and not another ctor overload?

Reply:
This is a private constructor only for `BlobContainerClient::GetBlobClient()`, for internal use only. Customers cannot directly use it.

We cannot have an options struct as a parameter for this constructor because in `BlobContainerClient`, we have `std::shared_ptr<Azure::Core::Http::HttpPipeline>` which contains all pipeline policies. We cannot separate them into per-operation policies and per-retry policies, thus cannot construct a `BlobClientOptions` struct.

Reply:
I'm going to proceed merging this PR, but we can still keep discussing about this.